### ISSUE TITLE
Change i64 seed to u64

### ISSUE THREAD

**Issue by vlovich** [ogxd/gxhash#78]
Breaking API change so would require a major bump, but all the other hashing libraries I've seen use u64. The i64 comes from the x64 intrinsic only working with signed integers (arm has `vdupq_n_u64`). Could the external API be changed to u64?

A more motivating reason to break the major version would be to make the function [more DOS resistant](#77) so if that's being done, maybe this could be done at the same time?

**Comment by ogxd** [comment]
Yes I agree I'd make more sense this way. Let's keep this as a side quest to DOS resistance improvements

**Comment by ogxd** [comment]
Done, will be released in v4
